anydex/wallet/ethereum/token/token_provider.py

```diff
@@ -0,0 +1,177 @@
+from anydex.wallet.ethereum.eth_provider import EtherscanProvider, AutoTestnetEthereumProvider, \
+    EthereumProvider, AutoEthereumProvider
+
+
+class TokenProvider(EthereumProvider):
+
+    def __init__(self, contract_address, abi, testnet=False):
+        """
+        Instantiate TokenProvider attributs.
+
+        :param contract_address: main token contract address
+        """
+        self.abi = abi  # read in default ERC20 Application Binary Interface
+        self._eth_provider = AutoTestnetEthereumProvider() if testnet else AutoEthereumProvider()
+        self.w3 = self._eth_provider.web3.w3
+        self.contract = self.w3.eth.contract(self.w3.toChecksumAddress(contract_address), abi=self.abi)
+        self._etherscan_provider = TokenEtherscanProvider(self.contract, testnet)
+
+    # @staticmethod

```

Comment:
Why is this commented? Should it be removed?

Reply:
Yeah ill remove it